app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,36 @@
+class PostsController < ApplicationController
+  expose_decorated :post
+
+  before_action :authorize_resource
+
+  def show
+  end
+
+  def new
+  end
+
+  def edit
+  end
+
+  def create
+    post = current_user.posts.create(post_params)
```

Comment:
You already exposed `post` instance.
```suggestion
    post.user = current_user
    post.save
```